tools/PI/DevHome.PI/Helpers/ExternalToolsHelper.cs

```diff
@@ -15,14 +17,44 @@ namespace DevHome.PI.Helpers;
 internal sealed class ExternalToolsHelper
 {
     private readonly JsonSerializerOptions serializerOptions = new() { WriteIndented = true };
-    private readonly ObservableCollection<ExternalTool> externalTools = [];
     private readonly string toolInfoFileName;
 
     public static readonly ExternalToolsHelper Instance = new();
 
     private static readonly ILogger _log = Log.ForContext("SourceContext", nameof(ExternalToolsHelper));
 
-    public ReadOnlyObservableCollection<ExternalTool> ExternalTools { get; set; }
+    private ObservableCollection<ExternalTool> allExternalTools = [];
+
+    // The ExternalTools menu shows all registered tools.
+    public ObservableCollection<ExternalTool> AllExternalTools
+    {
+        get => allExternalTools;
+        set
+        {
+            // We're assigning the collection once, and this also covers the case where we reassign it again:
+            // we need to unsubscribe from the old collection's events, subscribe to the new collection's events,
+            // and initialize the filtered collection.
+            if (allExternalTools != value)
+            {
+                if (allExternalTools != null)
+                {
+                    allExternalTools.CollectionChanged -= AllExternalTools_CollectionChanged;
+                }
+
+                allExternalTools = value;
+                if (allExternalTools != null)
+                {
+                    allExternalTools.CollectionChanged += AllExternalTools_CollectionChanged;
+                }
+
+                // Synchronize the filtered collection with this unfiltered one.
+                SynchronizeAllFilteredItems();
+            }
+        }
+    }
+
+    // The bar shows only the pinned tools.
+    public ObservableCollection<ExternalTool> FilteredExternalTools { get; private set; } = [];

```

Comment:
I *think* this should probably be in the BarWindow view model vs here... since the BarWindow would want a filtered view based on pinned, but other components using this (thinking about when we need input/output filters, extensions, etc) would want a different set of filtered tools.

Reply:
Yes, and that's a can of worms I want to defer to next round. ExternalToolsHelper should be split into a BarViewModel and other stuff.